### PR TITLE
Fix pyside2uic.UIParser on Python 3.9

### PR DESCRIPTION
**Pull request by n-elie**
Patch UIParser.readResources to make it works with Python 3.9. Without this patch, we get this error:

> Traceback (most recent call last):
  File "simple.py", line 12, in <module>
    MainWindowUI, MainWindowBase = uic.loadUiType(UI_FILE)
  File "\Miniconda3\envs\pysideads\lib\site-packages\qtpy\uic.py", line 271, in loadUiType
    compileUi(fd, code_stream, indent=0, from_imports=from_imports)
  File "\Miniconda3\envs\pysideads\lib\site-packages\pyside2uic\__init__.py", line 144, in compileUi
    winfo = ui_comp.compileUi(uifile, pyfile, from_imports)
  File "\Miniconda3\envs\pysideads\lib\site-packages\pyside2uic\Compiler\compiler.py", line 91, in compileUi
    w = self.parse(input_stream)
  File "\Miniconda3\envs\pysideads\lib\site-packages\pyside2uic\uiparser.py", line 882, in parse
    actor(elem)
  File "\Miniconda3\envs\pysideads\lib\site-packages\qtpy\uic.py", line 253, in readResources
    elem.getiterator = elem.iter
AttributeError: 'xml.etree.ElementTree.Element' object has no attribute 'getiterator'

> Methods getchildren() and getiterator() of classes ElementTree and Element in the ElementTree module have been removed. They were deprecated in Python 3.2. Use iter(x) or list(x) instead of x.getchildren() and x.iter() or list(x.iter()) instead of x.getiterator(). (Contributed by Serhiy Storchaka in bpo-36543.)

https://docs.python.org/3.9/whatsnew/3.9.html#removed

Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by n-elie** [comment]
This patch is also needed for 'pyside2-uic' command to work properly.
Without it, the following error occurs and import needed for resources loading and custom widgets are not written at the end of the file:

```An unexpected error occurred.
Check that you are using the latest version of PySide2 and report the error to
http://bugs.openbossa.org, including the ui file used to trigger the error.```

**Comment by n-elie** [comment]
@conda-forge-admin, please rerender

**Comment by n-elie** [comment]
Ready for review.
This PR also fixes build on Windows by relocating libpyside build to `sources/PySide2`. Not sure why it is needed
And it fixes a problem with `QToolButton.setMenu` which was not keeping a reference of the menu.

**Comment by n-elie** [comment]
Hello,

Is there something wrong with this PR? I personally need `pyside-uic` to work properly.

Regards

**Comment by jan-janssen** [comment]
> Hello,
> 
> Is there something wrong with this PR? I personally need `pyside-uic` to work properly.
> 
> Regards

It failed on Mac OS  X

**Comment by jan-janssen** [comment]
@conda-forge-admin, please rerender

**Comment by n-elie** [comment]
It was working well before I rerendered

**Comment by jan-janssen** [comment]
@conda-forge-admin, please restart ci

File: .ci_support/osx_64_python3.10.____cpython.yaml [hunk 1]
```diff
@@ -3,15 +3,15 @@ MACOSX_DEPLOYMENT_TARGET:
...
+- '12'
 channel_sources:
 - conda-forge
 channel_targets:
 - conda-forge main
 cxx_compiler:
 - clangxx
 cxx_compiler_version:
-- '11'
+- '12'
```
Comment (jan-janssen) [inline .ci_support/osx_64_python3.10.____cpython.yaml:14]: The part that changed seems to be the compiler version

**Comment by n-elie** [comment]
This time all checks have passed

**Comment by n-elie** [comment]
Thanks for your help

**Comment by hmaarrfk** [comment]
did this really fix anything? it seems like things are still broken.

**Comment by n-elie** [comment]
What is still broken exactly?

**Comment by hmaarrfk** [comment]
Sorry, i think i had an other conflict in one of my packages and the conda solver seemed to point me around in circles. Eventually, I was able to plug in the libmamba solver, but it wasn't as straightforward as I would have liked.

Thank you for your response.

**Pull request merged**